serialization/src/TorchForceProxy.cpp

```diff
@@ -55,9 +96,24 @@ void TorchForceProxy::serialize(const void* object, SerializationNode& node) con
 }
 
 void* TorchForceProxy::deserialize(const SerializationNode& node) const {
-    if (node.getIntProperty("version") != 1)
+    int storedVersion = node.getIntProperty("version");
+    if (storedVersion > 2)
         throw OpenMMException("Unsupported version number");
-    TorchForce* force = new TorchForce(node.getStringProperty("file"));
+    TorchForce* force;
+    if (storedVersion == 1) {
+        string fileName = node.getStringProperty("file");
+        force = new TorchForce(fileName);
+    } else if (storedVersion == 2) {
+        const string storedEncodedFile = node.getStringProperty("encodedFileContents", "");
+        if (storedEncodedFile.empty()) {
+            throw OpenMMException("Found and empty model file.");
+        }
```

Comment:
That message is a bit confusing, since there's no model file involved here.  I would remove the default value from `getStringProperty()`, which makes the `empty()` check unnecessary.  If this is version 2 or greater, the `encodedFileContents` property should be present.  If it isn't, the XML is invalid and you'll properly get an exception saying it's missing.

Reply:
Great! I was not aware of getStringProperty throwing.